.github/workflows/sbom.yaml

```diff
@@ -0,0 +1,37 @@
+name: sbom-build
```

Comment:
I like the model of one workflow for prs, one on merge and one on release. Should this be part of the build-test workflow? Why do we want this to run on pr?